src/components/Admin/AdminSearchForm.jsx

```diff
@@ -0,0 +1,163 @@
+/* eslint-disable camelcase */
+import qs from 'query-string';
+import React from 'react';
+import moment from 'moment';
+import PropTypes from 'prop-types';
+import { Form } from '@edx/paragon';
+
+import SearchBar from '../SearchBar';
+import { updateUrl } from '../../utils';
+
+class AdminSearchForm extends React.Component {
+  constructor(props) {
+    super(props);
+    const { location } = props;
+    const queryParams = qs.parse(location.search);
+    this.state = {
+      searchQuery: queryParams.search,
+      searchCourseQuery: queryParams.search_course,
+      searchDateQuery: queryParams.search_start_date,
+    };
+  }
+
+  componentDidUpdate(prevProps) {
+    const { location } = this.props;
+    if (location.search !== prevProps.location.search) {
+      // eslint-disable-next-line camelcase
+      const { search, search_course, search_start_date } = qs.parse(location.search);
+      const {
+        search: prevSearch,
+        search_course: prevSearchCourse,
```

Comment:
this is just convention/style wise: hmm to be consistent, could these also be camelCased? like searchCourse? or would that complicate work in sending over to the backend? I think some FE libs use a conversion pattern from snake to camel and back .. wondering if we want that here too

Reply:
Yeah, I thought about that. I'm parsing from the query string, so... I don't know. I don't think I want that here, I want it to be clear what's coming from the querystring.